modules/js-and-css/speculation-rules/helper.php

```diff
@@ -72,12 +72,17 @@ function plsr_get_speculation_rules() {
 				'and' => array(
 					// Prerender any URLs within the same site.
 					array(
-						'href_matches' => $prefixer->prefix_path_pattern( '/*' ),
+						'href_matches' => array( 'pathname' => $prefixer->prefix_path_pattern( '/*' ) ),
 					),
 					// Except for WP login and admin URLs.
 					array(
 						'not' => array(
-							'href_matches' => $href_exclude_paths,
+							'href_matches' => array_map(
+								static function ( string $path ): array {
+									return array( 'pathname' => $path );
+								},
+								$href_exclude_paths
+							),
```

Comment:
As you're saying in the PR description, this is potentially controversial. I'd like to understand whether this is truly needed? Wouldn't the escaping alone solve the problem (potentially being expanded to the full path string, rather than just the "path prefix")?

Maybe limiting this to paths makes sense, but at the moment this would prevent limiting by query parameters for instance, which is relevant for some WordPress sites (most importantly for the small subset of sites that doesn't use "pretty permalinks").

Reply:
https://github.com/whatwg/urlpattern/issues/210 has some more context if you're curious, but even if we do make this syntax legal it won't be accepted by Chromium for at least two milestones, so still worth this workaround.